### PR TITLE
Make add_importer and add_post_importer_plugin override existing importers.

### LINKED ISSUE #3826: Make add_importer and add_post_importer_plugin overriding existing importers
### Describe the project you are working on

Making a 3d multiplayer game and also working with the OMI group trying to implement audio emitters and other glTF2 extensions.

### Describe the problem or limitation you are having in your project

I want to implement importers that override existing importers.

See the sample project https://github.com/V-Sekai/godot-gltf-sandbox.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The `add_importer` and `add_post_importer_plugin` functions only work when its list of extensions are already not implemented by an existing importer like "glb" by the gltf importer.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Change the storage of the importers to be a vector and then add the new importer to the start of the list.

Usage of this proposal in Godot Engine 4.

```
@tool
extends EditorPlugin

var import_plugin = null

func _enter_tree():
 import_plugin = preload("res://addons/custom_importer/import_custom.gd").new()
 add_scene_format_importer_plugin(import_plugin)

func _exit_tree():
 remove_scene_format_importer_plugin(import_plugin)
 import_plugin = null
```

See pr.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Modifying the order of the importers cannot be done in gdscript as far as I know.

### Is there a reason why this should be core and not an add-on in the asset library?

I couldn't find a way to implement this in gdscript.

### PR DESCRIPTION
**Pull request by fire**
Fixes: https://github.com/godotengine/godot-proposals/issues/3826

### CODE REVIEW

File: editor/import/resource_importer_scene.cpp [hunk 1]
```diff
@@ -2110,3 +2110,23 @@ Node *EditorSceneFormatImporterESCN::import_scene(const String &p_path, uint32_t
...
+
+void ResourceImporterScene::add_post_importer_plugin(const Ref<EditorScenePostImportPlugin> &p_plugin) {
+	post_importer_plugins.reverse();
+	post_importer_plugins.push_back(p_plugin);
+	post_importer_plugins.reverse();
+}
+
+void ResourceImporterScene::remove_importer(Ref<EditorSceneFormatImporter> p_importer) {
+	importers.erase(p_importer);
+}
```
Comment (akien-mga) [inline editor/import/resource_importer_scene.cpp:2132]: See the context: this file has definitions for `ResourceImporterScene`, then `EditorSceneFormatImporterESCN`, and you're adding more `ResourceImporterScene` definitions after that. Those definitions should be moved up.

File: editor/import/resource_importer_scene.h [hunk 2]
```diff
@@ -153,7 +153,7 @@ VARIANT_ENUM_CAST(EditorScenePostImportPlugin::InternalImportCategory)
 class ResourceImporterScene : public ResourceImporter {
 	GDCLASS(ResourceImporterScene, ResourceImporter);
 
-	Set<Ref<EditorSceneFormatImporter>> importers;
+	mutable Vector<Ref<EditorSceneFormatImporter>> importers;
```
Comment (akien-mga) [inline editor/import/resource_importer_scene.h:156]: What does `mutable` do in this context?

Comment (fire) [reply editor/import/resource_importer_scene.h]: I don't know, all the other variables had it. Ain't going to need it. So removing.

File: editor/import/resource_importer_scene.cpp (hunk 1 above)
Comment (fire) [reply editor/import/resource_importer_scene.cpp]: Moved up.

File: editor/import/resource_importer_scene.h (hunk 2 above)
Comment (akien-mga) [reply editor/import/resource_importer_scene.h]: Well there must be a reason why the other ones have it, so someone needs to find out why :)

File: core/io/resource_importer.cpp [hunk 3]
```diff
@@ -463,3 +463,9 @@ void ResourceImporter::_bind_methods() {
 	BIND_ENUM_CONSTANT(IMPORT_ORDER_DEFAULT);
 	BIND_ENUM_CONSTANT(IMPORT_ORDER_SCENE);
 }
+
+void ResourceFormatImporter::add_importer(const Ref<ResourceImporter> &p_importer) {
+	importers.reverse();
+	importers.push_back(p_importer);
+	importers.reverse();
```
[the remark this answers is not in the document]
Comment (fire) [reply core/io/resource_importer.cpp]: I didn't see that function in my list. Thanks, it is cleaner.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**